pkg/chunk/cache/fifo_cache_test.go

```diff
@@ -2,100 +2,144 @@ package cache
 
 import (
 	"context"
-	"strconv"
+	"fmt"
 	"testing"
 	"time"
-	"unsafe"
 
 	"github.com/prometheus/client_golang/prometheus/testutil"
 	"github.com/stretchr/testify/assert"
 	"github.com/stretchr/testify/require"
 )
 
-const size = 10
-const overwrite = 5
-
-func TestFifoCache(t *testing.T) {
-	c := NewFifoCache("test1", FifoCacheConfig{Size: size, Validity: 1 * time.Minute})
+func TestFifoCacheEviction(t *testing.T) {
```

Comment:
Are you planning to add an eviction test for the memory limit configuration?

Reply:
The second test in this file tests both eviction and expiration for memory-based eviction, but it is a good idea to run these tests for both cases. I'll make a change.